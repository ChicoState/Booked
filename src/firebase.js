import * as firebase from 'firebase';
import 'firebase/auth';

var config = {
  apiKey: "AIzaSyAbrPremBQusjYoKqKfTcVHwxuw-Q8ikHw",
  authDomain: "booked-8586e.firebaseapp.com",
  databaseURL: "https://booked-8586e.firebaseio.com",
  projectId: "booked-8586e",
  storageBucket: "booked-8586e.appspot.com",
  messagingSenderId: "24720035765",
  appId: "1:24720035765:web:3d57fd5c840b8100c353fc",
  measurementId: "G-20QKEY7V1D"
};

firebase.initializeApp(config);

// var provider = new firebase.auth.GoogleAuthProvider();
// //var user = firebase.auth().currentUser;
// var name, email;
//
//
// firebase.auth().useDeviceLanguage();
// firebase.auth().onAuthStateChanged(function(user){
//
// 	if(user != null){
// 		name = user.displayName;
// 		email = user.email;
// 		alert(name + ': ' + email);
// 	}
// 	else{
// 		alert('Working on Redirect for initial log in');
// 		firebase.auth.signinWithRedirect(provider);
// 		firebase.auth().getRedirectResult().then(function(result) {
// 			if (result.credential) {
// 				//This gives you a Google Access Token.
// 				//var token = result.credential.accessToken;
// 			}
// 			user = result.user;
// 		}).catch(function(error) {
// 				// Handle Errors here.
// 				var errorCode = error.code;
// 				var errorMessage = error.message;
// 				// The email of the user's account used.
// 				var errorEmail = error.email;
// 				// The firebase.auth.AuthCredential type that was used.
// 				var credential = error.credential;
//
// 				//Output line for error variables to silence unused var warnings
// 				alert(errorCode + errorMessage + errorEmail + credential);
// 		});
// 	}
// });
