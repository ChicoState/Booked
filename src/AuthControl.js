import React from 'react';
//import * as firebase from "firebase/app";
import firebase from "firebase";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import { withRouter } from 'react-router-dom';
//import 'firebase/auth';
//import app from './firebase';

export default class AuthControl extends React.Component {
//class AuthControl extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isSignedIn : false};
  }
  uiConfig = {
    signInFlow: "popup",
    signInOptions: [
      firebase.auth.GoogleAuthProvider.PROVIDER_ID
    ],
    callback: {
      signInSuccess: () => false
    }
  }

  componentDidMount = () => {
    new Promise((resolve) => {
    firebase.auth().onAuthStateChanged(user => {
      resolve(user)
      this.setState({ isSignedIn: !!user})
      console.log("user", user)
    })
  })
  }

  render() {
    return (
      <div>
      {this.state.isSignedIn ?
      (
        <span>
          <button id="outty" onClick={()=>firebase.auth().signOut()}>Sign out!</button>
          <h1>Welcome {firebase.auth().currentUser.displayName}</h1>
          <img alt="profile" src={firebase.auth().currentUser.photoURL}/>
        </span>
      )
      :
      (
        <StyledFirebaseAuth
          uiConfig={this.uiConfig}
          firebaseAuth={firebase.auth()}
        />
      )
      }
      </div>
    )
  }
}

//export default withRouter(AuthControl);
