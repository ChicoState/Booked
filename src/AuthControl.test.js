import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
//import configureMockStore from 'redux-mock-store';
//import thunk from 'redux-thunk';
//import { Provider } from 'react-redux';
import AuthControl from './AuthControl';
import firebase from 'firebase';

// const mockStore = configureMockStore([thunk]);
// const store = mockStore();
// const FirebaseAuthProps = {};

jest.mock('firebase', () => {
  return {
    auth: () => {
      return {
        onAuthStateChanged: jest.fn(cb => cb({ name: 'Testy' }))
      };
    },
  };
});


jest.mock('firebase', () => {
  const auth = jest.fn();
  const mAuth = { signInWithRedirect: jest.fn() };
  auth.GoogleAuthProvider = jest.fn();
  auth.Auth = jest.fn(() => mAuth);
  return { auth };
});

describe('<AuthControl />', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
const wrapper = mount(<AuthControl />);
if(wrapper.find('outty') > 0){
  wrapper.find('outty').simulate("click");
}

it('renders without crashing', () => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  const div = document.createElement('div');
  ReactDOM.render(<AuthControl />, div);
});

test('compare output', () => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  const component = renderer.create(
    <AuthControl />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//   const wrapper = mount(<AuthControl />)
//   wrapper.find('button').simulate('click');
//   expect(firebase.auth.GoogleAuthProvider).toBeCalledTimes(1);
// expect(firebase.auth).toBeCalledTimes(1);
// expect(firebase.auth().signInWithRedirect).toBeCalledTimes(1);

});
