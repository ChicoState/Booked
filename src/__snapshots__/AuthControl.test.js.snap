// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthControl /> compare output 1`] = `
<div>
  <div
    id="firebaseui_container"
  />
</div>
`;
